src/winter_onboarding_2021/fleet_management_service/middleware.clj

```diff
@@ -41,17 +41,22 @@
                                             (timbre/log level throwable message)
                                             (timbre/log level message)))})))
 
+(defn user-from-session-id [session-id]
+  (try
+    (-> session-id
+        java.util.UUID/fromString
+        session/user-session
+        first)
+    (catch Exception _ nil)))
```

Comment:
Why are we silently catching this exception?

Reply:
Earlier we were expecting whatever we will receive in session-id in request is a uuid. But thats not always the case now since I am removing the session-id after logout. So, for the upcoming request the session-id is empty (which is not an uuid) `java.util.UUID/fromString` is raising an exception. So, I am using a wrapper function `user-from-session-id` which will return a valid session-id (if it is) or else nil if the passed id is not an uuid.

Reply:
P. S  We are returning `nil` if `java.util.UUID/fromString` raises an exception if the session-id is not an uuid.

Reply:
[update] I have removed this function, since I was checking if the session-id is "" or  `nil` ,(while destroying session-id, the value was set to "" where the session-id key still existed). Now I am able to remove session-id from the cookie by setting max-age to 0 which remove entire key value pair of session-id in response and consecutive requests. Hence we don't need to check for nil or "".